plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -0,0 +1,51 @@
+from operator import itemgetter
+
+import insightconnect_plugin_runtime
+from typing import Dict, List
+from insightconnect_plugin_runtime.helper import get_time_hours_ago
+from dateutil.parser import parse
+
+from .schema import MonitorSiemLogsInput, MonitorSiemLogsOutput, MonitorSiemLogsState, Component
+from ...util.event import EventLogs
+
+
+class MonitorSiemLogs(insightconnect_plugin_runtime.Task):
+    NEXT_TOKEN = "next_token"  # nosec
+    HEADER_NEXT_TOKEN = "mc-siem-token"  # nosec
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_siem_logs",
+            description=Component.DESCRIPTION,
+            input=MonitorSiemLogsInput(),
+            output=MonitorSiemLogsOutput(),
+            state=MonitorSiemLogsState(),
+        )
+
+    def run(self, params={}, state={}) -> (List[Dict], Dict):
+        if not state:
+            self.logger.info("First run")
+        else:
+            self.logger.info("Subsequent run")
+            params["token"] = state.get(self.NEXT_TOKEN)
+
+        has_more_pages = False
+
+        while True:
+            output, headers = self.connection.client.get_siem_logs(params)
+            header_next_token = headers.get(self.HEADER_NEXT_TOKEN)
+            params["token"] = header_next_token
+
+            output = [EventLogs(datetime=parse(event.pop("datetime"), ignoretz=True), data=event) for event in output]
```

Comment:
If the timezone is ignored, does this tie in correctly when comparison is made against get_time_hours_ago (is this in utc time?)?

Reply:
yes, we pass our actual time in the authorization header